helper/threads.c

```diff
@@ -118,6 +122,25 @@ static int create_process(odph_thread_t *thread, int cpu)
 		return -2;
 	}
 
+	if (stack_size) {
+		struct rlimit rlimit;
+
+		if (getrlimit(RLIMIT_STACK, &rlimit)) {
+			ODPH_ERR("getrlimit() failed: %s\n", strerror(errno));
+			return -3;
+		}
+
+		rlimit.rlim_cur = stack_size;
+
+		if (rlimit.rlim_cur > rlimit.rlim_max)
+			rlimit.rlim_max = rlimit.rlim_cur;
+
+		if (setrlimit(RLIMIT_STACK, &rlimit)) {
```

Comment:
An unprivileged user is only able to set `rlim_cur` (up to current `rlim_max`). Maybe it would be good to add a specific error print for this case (EPERM).

Reply:
This prints something like: "setrlimit() failed: Operation not permitted". Isn't that enough?

Reply:
After another look, I think it's best if we don't try to change rlim_max here, it should come from the system administrator (or equivalent). So I removed this in v3.